streams/src/main/java/org/apache/kafka/streams/state/internals/InMemoryKeyValueStore.java

```diff
@@ -98,9 +96,15 @@ public synchronized void put(final Bytes key, final byte[] value) {
     }
 
     @Override
-    public void putAll(final List<KeyValue<Bytes, byte[]>> entries) {
+    public synchronized void putAll(final List<KeyValue<Bytes, byte[]>> entries) {
         for (final KeyValue<Bytes, byte[]> entry : entries) {
-            put(entry.key, entry.value);
+            // intended to duplicate codes in #put, to avoid continuously lock/unlock cost
```

Comment:
We should not duplicate code, but instead extract an internal `private` helper method `putInternal` that can be called by `put` and `putAll`